client/txn.go

```diff
@@ -78,18 +78,37 @@ type Txn struct {
 	haveEndTxn   bool // True if there was an explicit EndTransaction
 }
 
-func newTxn(db DB, depth int) *Txn {
+func newTxn(db DB) *Txn {
 	txn := &Txn{
 		db:      db,
 		wrapped: db.Sender,
 	}
 	txn.db.Sender = (*txnSender)(txn)
 
-	file, line, fun := caller.Lookup(depth + 1)
+	// Caller's caller.
+	file, line, fun := caller.Lookup(2)
 	txn.txn.Name = fmt.Sprintf("%s:%d %s", file, line, fun)
 	return txn
 }
 
+// NewTxn returns a new txn.
+func NewTxn(db DB) *Txn {
```

Comment:
Hmmm, I see what you are saying. The name is currently the location of the caller (transaction code snippet). Since in SQL this will be some remote application, the application will need to bundle some information that we can use here. The BEGIN statement doesn't support this, but I suppose we could have BEGIN support a name. Let me open an issue about this.